src/util/std_expr.cpp

```diff
@@ -114,6 +113,15 @@ static void build_object_descriptor_rec(
 
     build_object_descriptor_rec(ns, tc.op(), dest);
   }
+  else if(const auto deref = expr_try_dynamic_cast<dereference_exprt>(expr))
```

Comment:
Maybe briefly comment why this isn't a task for the simplifier?

Reply:
This is something the simplifier should cover, but this piece of code should be usable irrespective of whether an expression has been simplified or not.